lib/Containers/FlatHashSet.h

```diff
@@ -25,11 +25,36 @@
 #include <absl/container/flat_hash_set.h>
 
 namespace arangodb::containers {
+namespace detail {
+
+template<typename T>
+char isCompleteHelper(char (*)[sizeof(T)]);
+
+template<typename T>
+int isCompleteHelper(...);
+
+template<size_t SizeofT, typename T>
+constexpr bool SetSizeofChecker() noexcept {
+  if constexpr (sizeof(decltype(isCompleteHelper<T>(nullptr))) ==
+                sizeof(char)) {
+    static_assert(sizeof(T) <= SizeofT,
+                  "For large T better to use NodeHashSet. "
+                  "If you really sure what do you do,"
+                  " please use absl::flat_hash_set directly.");
+    return sizeof(T) <= SizeofT;
+  } else {
+    return true;
+  }
+}
+
+}  // namespace detail
 
 template<class T, class Hash = typename absl::flat_hash_set<T>::hasher,
          class Eq = typename absl::flat_hash_set<T, Hash>::key_equal,
          class Allocator =
-             typename absl::flat_hash_set<T, Hash, Eq>::allocator_type>
+             typename absl::flat_hash_set<T, Hash, Eq>::allocator_type,
+         // TODO(MBkkt) After additional benchmarks make SizeofT bigger
+         class = std::enable_if<detail::SetSizeofChecker<32, T>(), void>>
```

Comment:
To me this looks like a personal TODO note;

 I cannot make any sense of this comment and wouldn't be able to action it if I found it (other than deleting it).

I also do not understand what this optimisation would achieve.

Reply:
About "personal" part of todo, I trying to make all my todo personal, because I think it makes it better

I think our codestyle doesn't have something against it

Reply:
Again: what is any other person who finds this note supposed to do with it?

They don't know what benchmarks, they don't know what this supposed optimisation does.

You explained it above, why is this explanation not in the code?

If you want a todo like this, maybe make a benchmark file somewhere that you can refer to so that another person who finds this todo can do something about it.

If you cannot add this kind of information and most of it is just in your head, it is of no use to anyone else. So leave the todonote somewhere in your personal files.